lib/template-loader.php

```diff
@@ -396,37 +319,6 @@ function gutenberg_strip_php_suffix( $template_file ) {
 	return preg_replace( '/\.php$/', '', $template_file );
 }
 
-/**
- * Extends default editor settings to enable template and template part editing.
- *
- * @param array $settings Default editor settings.
- *
- * @return array Filtered editor settings.
- */
-function gutenberg_template_loader_filter_block_editor_settings( $settings ) {
-	global $post;
-
-	if ( ! $post ) {
-		return $settings;
-	}
-
-	// If this is the Site Editor, auto-drafts for template parts have already been generated
-	// through `filter_rest_wp_template_part_query`, when called via the REST API.
-	if ( isset( $settings['editSiteInitialState'] ) ) {
-		return $settings;
-	}
-
-	// Otherwise, create template part auto-drafts for the edited post.
-	$post = get_post();
-	foreach ( parse_blocks( $post->post_content ) as $block ) {
-		create_auto_draft_for_template_part_block( $block );
-	}
-
-	// TODO: Set editing mode and current template ID for editing modes support.
-	return $settings;
-}
-add_filter( 'block_editor_settings', 'gutenberg_template_loader_filter_block_editor_settings' );
```

Comment:
I believe this filter was necessary in the early stages when we bootstraped everything from the server?

Reply:
I think we need this behavior for standalone template post editing (From appearance->templates). My understanding is that this makes sure that all of the template parts are in the database for the template we wish to edit.

I think we can remove it if the query for a template part from the post editor also adds them to the database if they haven't been created yet

Reply:
> his PR starts with the following assumption: The site editor always fetches template parts using REST API.
Based on this premise, why not just create these auto drafts on all GET requests to template parts? It's important to note that it's close to how it works currently but right now, it's hidden behind a number of function calls and is not consistent.

I have this in the description of the PR. Template parts are loaded through the REST API which does the syncing, I think we're covered.